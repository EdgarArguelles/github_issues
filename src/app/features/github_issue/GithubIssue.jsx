import './GithubIssue.scss';
import React from 'react';
import PropTypes from 'prop-types';
import IssueLabel from './issue_label/IssueLabel';

const GithubIssue = ({issue}) => {
    return (
        <div className="github-issue">
            <div>
                <b>‚ùó{issue.title}</b>
                {issue.labels.map(label => <IssueLabel key={`label-${label.id}`} label={label}/>)}
            </div>
            <div className="subtitle">
                #{issue.number} opened by {issue.user.login}
                <img src={issue.user.avatar_url} alt={issue.user.login} className="user-avatar"/>
            </div>
        </div>
    );
};

GithubIssue.propTypes = {
    issue: PropTypes.object.isRequired,
};

export default React.memo(GithubIssue);