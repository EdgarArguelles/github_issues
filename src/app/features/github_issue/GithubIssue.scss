.github-issue {
  .subtitle {
    font-style: italic;
    color: grey;
    font-size: 0.8rem;
    text-align: right;
    margin: 5px 30px 0 0;

    .user-avatar {
      margin-left: 15px;
      max-width: 20px;
      border-radius: 50%;
      position: absolute;
    }
  }
}