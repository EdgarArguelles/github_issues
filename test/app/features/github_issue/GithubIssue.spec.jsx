import renderer from 'react-test-renderer';
import React from 'react';
import GithubIssue from 'app/features/github_issue/GithubIssue';

describe('Features -> GithubIssue', () => {
    let component, wrapper;

    const mountComponent = props => {
        wrapper = renderer.create(<GithubIssue {...props}/>);
        component = wrapper.root.find(el => el.type.name === 'GithubIssue');
    };

    describe('load', () => {
        it('should have all props', () => {
            mountComponent({
                issue: {
                    title: 'title test',
                    number: 5,
                    user: {login: 'login test'},
                    labels: [{id: 1, color: 'red', name: 'A'}, {id: 2, color: 'blue', name: 'B'}],
                },
            });

            expect(component.props.issue).toStrictEqual({
                title: 'title test',
                number: 5,
                user: {login: 'login test'},
                labels: [{id: 1, color: 'red', name: 'A'}, {id: 2, color: 'blue', name: 'B'}],
            });
        });
    });

    describe('snapshot', () => {
        it('should render GithubIssue with props', () => {
            mountComponent({
                issue: {
                    title: 'title test',
                    number: 5,
                    user: {login: 'login test'},
                    labels: [{id: 1, color: 'red', name: 'A'}, {id: 2, color: 'blue', name: 'B'}],
                },
            });

            expect(wrapper.toJSON()).toMatchSnapshot();
        });
    });
});