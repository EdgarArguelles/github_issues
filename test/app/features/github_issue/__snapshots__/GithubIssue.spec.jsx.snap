// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features -> GithubIssue snapshot should render GithubIssue with props 1`] = `
<div
  className="github-issue"
>
  <div>
    <b>
      ‚ùó
      title test
    </b>
    <span
      className="issue-label"
      style={
        Object {
          "backgroundColor": "#red26",
          "borderColor": "#red",
        }
      }
    >
      A
    </span>
    <span
      className="issue-label"
      style={
        Object {
          "backgroundColor": "#blue26",
          "borderColor": "#blue",
        }
      }
    >
      B
    </span>
  </div>
  <div
    className="subtitle"
  >
    #
    5
     opened by 
    login test
    <img
      alt="login test"
      className="user-avatar"
    />
  </div>
</div>
`;
